src/zenml/artifacts/load_directory_materializer.py

```diff
@@ -0,0 +1,93 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Only-load materializer for directories."""
+
+import os
+import tempfile
+from pathlib import Path
+from typing import Any, ClassVar, Tuple, Type
+
+from zenml.enums import ArtifactType
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+
+class PreexistingArtifactPath(type(Path())):  # type: ignore[misc]
+    """Simple wrapper to protect from misuse."""
+
+    pass
+
+
+class PreexistingDataMaterializer(BaseMaterializer):
+    """Materializer to load directories from the artifact store.
+
+    This materializer is very special, since it do not implement save
+    logic at all. The save of the data to some URI inside the artifact store
+    shall happen outside and is in user's responsibility.
+
+    This materializer solely supports the `link_existing_data_as_artifact` function.
+    """
+
+    ASSOCIATED_TYPES: ClassVar[Tuple[Type[Any], ...]] = (
+        PreexistingArtifactPath,
+    )
+    ASSOCIATED_ARTIFACT_TYPE: ClassVar[ArtifactType] = ArtifactType.DATA
+
+    def load(self, data_type: Type[Any]) -> Any:
+        """Copy the artifact file(s) to a local temp directory.
+
+        Args:
+            data_type: Unused.
+
+        Returns:
+            Path to the local directory that contains the artifact files.
+        """
+        directory = tempfile.mkdtemp(prefix="zenml-artifact")
+        if fileio.isdir(self.uri):
+            self._copy_directory(src=self.uri, dst=directory)
+            return PreexistingArtifactPath(directory)
+        else:
+            dst = os.path.join(directory, os.path.split(self.uri)[-1])
+            fileio.copy(src=self.uri, dst=dst)
+            return PreexistingArtifactPath(dst)
+
+    def save(self, data: Any) -> None:
+        """Store the directory in the artifact store.
+
+        Args:
+            data: Path to a local directory to store.
+
+        Raises:
+            NotImplementedError: Always
+        """
+        raise NotImplementedError
```

Comment:
I still think this should have at least a short error message explaining to not use this materializer to save things, just in case

Reply:
Added